File: VonexDealer.Vue/ClientApp/models/Address.js
export class Address {
    constructor({ addressID = 0,
        streetAddress = null,
        suburb = null,
        state = null,
        postCode = null,
        customerID = 0} = {})

    {
        this.addressID = addressID;
        this.streetAddress = streetAddress;
        this.suburb = suburb;
        this.state = state;
        this.postCode = postCode;
        this.customerID = customerID;
    }
}

export function createAddress(data) {
    return Object.freeze(new Address(data));
}

File: VonexDealer.API/Models/Order/Address.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace VonexDealer.API.Models.Order
{
    /// <summary>
    /// this data should be collected from TotalCheck verification process - TCAddress
    /// </summary>
    public class Address
    {
        public Int64 AddressID { get; set; }
        public string StreetAddress { get; set; }
        public string Suburb { get; set; }
        public string State { get; set; }
        public int PostCode { get; set; }
        public Int64? CustomerID { get; set; }
        [NotMapped]
        public string FormattedAddress
        {
            get
            {
                return $"{StreetAddress}, {Suburb}, {State}, {PostCode}";
            }
        }
    }
}
